runtime/lua/vim/lsp/util.lua

```diff
@@ -688,6 +699,19 @@ do
       api.nvim_buf_set_virtual_text(bufnr, diagnostic_ns, line, virt_texts, {})
     end
   end
+  function M.buf_diagnostics_signs(bufnr, diagnostics)
+    for _, diagnostic in ipairs(diagnostics) do
+      if diagnostic.severity == protocol.DiagnosticSeverity.Error then
+        vim.fn.sign_place(0, sign_ns, 'LspDiagnosticsErrorSign', bufnr, {lnum=(diagnostic.range.start.line+1)})
+      elseif diagnostic.severity == protocol.DiagnosticSeverity.Warning then
+        vim.fn.sign_place(0, sign_ns, 'LspDiagnosticsWarningSign', bufnr, {lnum=(diagnostic.range.start.line+1)})
+      elseif diagnostic.severity == protocol.DiagnosticSeverity.Information then
+        vim.fn.sign_place(0, sign_ns, 'LspDiagnosticsInformationSign', bufnr, {lnum=(diagnostic.range.start.line+1)})
+      elseif diagnostic.severity == protocol.DiagnosticSeverity.Hint then
+        vim.fn.sign_place(0, sign_ns, 'LspDiagnosticsHintSign', bufnr, {lnum=(diagnostic.range.start.line+1)})
+      end
```

Comment:
the call remains the same apart from the hl name. Either assign the name into avariable or create a function to map the level to the highlight name or just change the if.
`vim.fn.sign_place(0, sign_ns, <NAME>, bufnr, {lnum=(diagnostic.range.start.line+1)})` should appear once.